src/pilot/discover_nodes/discover_nodes.py

```diff
@@ -501,14 +485,8 @@ def is_idrac(client):
         # Ensure the libraries' loggers are re-enabled.
         requests_logger.disabled = False
         dracclient_wsman_logger.disabled = False
-
-    if doc is None:
-        return False
-
-    return dracclient.utils.get_wsman_resource_attr(
-        doc,
-        DCIM_iDRACCardView,
-        'DeviceDescription') == 'iDRAC'
+    if is_ready in (True, False):
```

Comment:
The above line is not needed.  If any exception occurs, it will be caught above at line 496.  If we made it this far in the code, then the server responded to the is_idrac_ready check, so we can just "return True" here.

Reply:
Removed extra if statement and updated the code returning "True" value.